lua/ocaml/init.lua

```diff
@@ -138,6 +137,94 @@ function ocaml.phrase(dir)
   end)
 end
 
+function ocaml.find_identifier_def(identifier)
+  with_server(function(client)
+    local row, col = unpack(vim.api.nvim_win_get_cursor(0))
+    local params = {
+      uri = vim.uri_from_bufnr(0),
+      command = "locate",
+      args = {
+        "-position",
+        row + 1 .. ":" .. col,
+        "-prefix",
+        identifier,
+        "-look-for",
+        "implementation",
+      },
+      resultAsSexp = false,
+    }
+    local result = client.request_sync("ocamllsp/merlinCallCompatible", params, 1000)
+    if not (result and result.result) then
+      vim.notify("No definition for identifier " .. identifier .. ".", vim.log.levels.WARN)
+      return
+    end
+    local data = result.result.result
+    local ok, parsed = pcall(vim.fn.json_decode, data)
+    if not ok or not parsed.value or not parsed.value.pos then
+      vim.notify("Invalid response from server.", vim.log.levels.ERROR)
+      return
+    end
+    local file = parsed.value.file
+    local line = parsed.value.pos.line
+    local col = parsed.value.pos.col
+    vim.cmd.split(file)
+    vim.api.nvim_win_set_cursor(0, { line, col })
+  end)
+end
+
+function ocaml.find_identifier_decl(identifier)
+  with_server(function(client)
+    local row, col = unpack(vim.api.nvim_win_get_cursor(0))
+    local params = {
+      uri = vim.uri_from_bufnr(0),
+      command = "locate",
+      args = {
+        "-position",
+        row + 1 .. ":" .. col,
+        "-prefix",
+        identifier,
+        "-look-for",
+        "interface",
+      },
+      resultAsSexp = false,
+    }
+    local result = client.request_sync("ocamllsp/merlinCallCompatible", params, 1000)
+    if not (result and result.result) then
+      vim.notify("No declaration for identifier " .. identifier .. ".", vim.log.levels.WARN)
+      return
+    end
+    local data = result.result.result
+    local ok, parsed = pcall(vim.fn.json_decode, data)
+    if not ok or not parsed.value or not parsed.value.pos then
+      vim.notify("Invalid response from server.", vim.log.levels.ERROR)
+      return
+    end
+    local file = parsed.value.file
+    local line = parsed.value.pos.line
+    local col = parsed.value.pos.col
+    vim.cmd.split(file)
+    vim.api.nvim_win_set_cursor(0, { line, col })
+  end)
+end
```

Comment:
since the difference between the two function is the `look-for` value, WDYT about mutualize that?